### PR TITLE
[risk=low][RW-7280] Apply eslint rule @typescript-eslint/prefer-optional-chain

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/analysis/runtime-panel.tsx [hunk 1]
```diff
@@ -1532,8 +1532,7 @@ const RuntimePanel = fp.flow(
     const pdSize = pdExists ? persistentDisk.size : defaultDiskSize;
     const initialAutopauseThreshold =
       existingRuntime.autopauseThreshold || DEFAULT_AUTOPAUSE_THRESHOLD_MINUTES;
-    const gpuConfig =
-      gceConfig && gceConfig.gpuConfig ? gceConfig.gpuConfig : null;
+    const gpuConfig = gceConfig?.gpuConfig ? gceConfig.gpuConfig : null;
```
Comment (jmthibault79) [inline ui/src/app/pages/analysis/runtime-panel.tsx:1535]: can maybe do a little better here?

File: ui/src/app/pages/data/cohort-review/cohort-definition.component.tsx [hunk 2]
```diff
@@ -296,49 +296,47 @@ export class CohortDefinition extends React.Component<
...
-                  <div style={styles.groupBackground}>
-                    {item.map((param, p) => (
-                      <React.Fragment key={p}>
-                        {p > 0 && <div>OR</div>}
-                        <div>
-                          {param.map((crit, c) => (
-                            <React.Fragment key={c}>
-                              {c > 0 && (
-                                <React.Fragment>
-                                  {crit.domain === Domain.PERSON && (
```
Comment (jmthibault79) [inline ui/src/app/pages/data/cohort-review/cohort-definition.component.tsx:323]: not sure why the linter did anything here, but I took the opportunity to turn it into a ternary

File: ui/src/app/pages/analysis/runtime-panel.tsx (hunk 1 above)
Comment (jmthibault79) [reply ui/src/app/pages/analysis/runtime-panel.tsx:1535]: nah, this is used in a few places and I'm not sure of the implications of undefined vs. null here

File: ui/src/app/pages/data/cohort-review/cohort-definition.component.tsx (hunk 2 above)
Comment (dolbeew) [reply ui/src/app/pages/data/cohort-review/cohort-definition.component.tsx:323]: Nice, thanks.

**Review by dolbeew — APPROVED** [review]

**Pull request merged**